src/app/notifications/reducer.spec.ts

```diff
@@ -70,11 +72,61 @@ describe('notifications reducer', () => {
       (state) => reducer(state, actions.dismissNotification(acceptCookiesNotification))
     )(initialState);
 
-    expect(newState).not.toContainEqual(actions.acceptCookies());
+    expect(newState.modalNotifications).not.toContainEqual(actions.acceptCookies());
+    expect(newState.toastNotifications).toEqual([]);
   });
 
   it('reduces acceptCookies', () => {
     const newState = reducer(initialState, actions.acceptCookies());
-    expect(newState).toContainEqual(actions.acceptCookies());
+    expect(newState.modalNotifications).toContainEqual(actions.acceptCookies());
+    expect(newState.toastNotifications).toEqual([]);
+  });
+
+  describe('toast notifications', () => {
+    it('reduces toasts', () => {
+      const newState = flow(
+        (state) => reducer(state, actions.addToast('mytoast')),
+        (state) => reducer(state, actions.addToast('myothertoast'))
+      )(initialState);
+
+      expect(newState.modalNotifications).toEqual([]);
+      expect(newState.toastNotifications).toContainEqual(expect.objectContaining({message: 'mytoast'}));
+      expect(newState.toastNotifications).toContainEqual(expect.objectContaining({message: 'myothertoast'}));
+    });
+
+    it('refreshes the timestamp if a toast with the same message appears', async() => {
+      const newState = reducer(initialState, actions.addToast('mytoast'));
+      const toast = newState.toastNotifications.find((notification) => notification.message === 'mytoast');
+
+      if (!toast) {
+        return expect(toast).toBeTruthy();
+      }
+      await new Promise((res) => setTimeout(res, 10));
```

Comment:
Maybe you could try to mock the return value for `action.addToast(...)` to return another timestamp, this way we will avoid blocking a thread

Reply:
I'm just gonna reduce the timeout, the less mocks the better.

Reply:
Maybe this can be used by using `jest.useFakeTimers();`? 
The thing is that we should try to make our unit tests as sync as possible, and `setTimeout` set even to `5`ms will block a thread for **at least** 5 ms. That sounds like a small amount of time but if we keep doing so we can end up a bigger and bigger amount